src/global.cpp

```diff
@@ -36,27 +36,82 @@ int symPACK_Rank(int * rank){
 }
 
 
+#ifdef CUDA_MODE
+extern "C"
+void compute_device_alloc_size() {
+  int n_gpus = upcxx::gpu_default_device::device_n();
+  if (n_gpus==0) {
+    std::cerr<<"Error: found no CUDA devices"<<std::endl;
+    abort();
+  }
+
+  MPI_Comm shmcomm;
+  int tasks_per_node;
+  MPI_Comm_split_type(MPI_COMM_WORLD, MPI_COMM_TYPE_SHARED, 0, MPI_INFO_NULL, &shmcomm);
+  MPI_Comm_size(shmcomm, &tasks_per_node);
```

Comment:
The UPC++ idiom for this:

```suggestion
  int tasks_per_node = upcxx::local_team().rank_n();
```